components/lists/sort/controller.tsx

```diff
@@ -7,24 +9,32 @@ import { CustomDropdownItem, SortOption } from "types/components";
 interface ListSortProps {
   defaultOptionIndex?: number;
   options: SortOption[];
+  asSelect?: boolean;
 }
 
 export default function ListSort({
   defaultOptionIndex = 0,
-  options
+  options,
+  asSelect,
 }: ListSortProps) {
   const router = useRouter();
+
+  const [selectedIndex, setSelectedIndex] = useState<number>();
+
   const { sortBy, order } = router.query;
 
   function handleSelectChange(newValue) {
+    setSelectedIndex(options?.findIndex(option => option.sortBy === newValue.sortBy && 
+                                        option.order === newValue.order));
+
     const query = {
       ...router.query,
       sortBy: newValue.sortBy,
       order: newValue.order,
       page: "1"
     };
 
-    router.push({ pathname: `${router.pathname}`, query }, router.asPath, { shallow: false, scroll: false });
+    router.push({ pathname: router.pathname, query }, router.asPath, { shallow: false, scroll: false });
```

Comment:
if we route.push with `shallow: false` the page will reload, thus the `setSelectedIndex` will be replaced with the one from the refreshed page

Reply:
@moshmage the page won't be reloaded, the `shallow` property only prevents the `getServerSideProps` to be executed, with `shallow: false` I'm ensuring that will run with the new `sortBy` and `order` values